pkg/frontend/querymiddleware/shard_by_series_base.go

```diff
@@ -72,7 +72,7 @@ func (s *shardBySeriesBase) shardBySeriesSelector(ctx context.Context, spanLog *
 	resp, err := doShardedRequests(ctx, reqs, s.upstream)
 	if err != nil {
 		if errors.Is(err, errShardCountTooLow) {
-			return nil, apierror.New(apierror.TypeBadData, fmt.Errorf("%w: try increasing the requested shard count", err).Error())
```

Comment:
Doesn't this change the status code for all errors, regardless what caused them?

Reply:
What do you mean by "all errors"? This only applies to errors for which `errors.Is(err, errShardCountTooLow)` returns true. Am I missing something? 🤔

Reply:
🙈 you're right. That's why I shouldn't do code reviews from my phone while waiting for a coffee

LGTM